pandas/core/resample.py

```diff
@@ -856,10 +856,32 @@ def size(self):
         # It's a special case as higher level does return
         # a copy of 0-len objects. GH14962
         result = self._downsample("size")
-        if not len(self.ax) and isinstance(self._selected_obj, ABCDataFrame):
+        if not len(self.ax):
             from pandas import Series
 
-            result = Series([], index=result.index, dtype="int64")
+            if self._selected_obj.ndim == 1:
```

Comment:
so we already have count and size defined in groupby on the appropriate objects, see pandas/core/groupby/generic.py

but now we end up re-writing them here. Can you see if there is a way to combine these?

Reply:
Yea agreed on this comment. This will add a third `count` implementation for classes that subclass `_GroupBy` (`SeriesGroupBy` and `DataFrameGroupBy` each have their own)

Do you see any way to consolidate?

Reply:
I'm not sure there's much to share between this implementation and the other ones.

Reply:
Hmm OK. Would definitely take this as a follow up to refactor - having three different implementations for objects deriving from the same base class is less than ideal

Reply:
pls combine these; we don't need both paths; these are the same. if some tests need changing because the behavior changed, then pls do this.